numpy/_core/tests/test_stringdtype.py

```diff
@@ -809,18 +810,38 @@ def test_ufunc_multiply(dtype, string_list, other, other_dtype, use_out):
             other * arr
 
 
-def test_datetime_cast(dtype):
-    a = np.array(
-        [
-            np.datetime64("1923-04-14T12:43:12"),
-            np.datetime64("1994-06-21T14:43:15"),
-            np.datetime64("2001-10-15T04:10:32"),
-            np.datetime64("NaT"),
-            np.datetime64("1995-11-25T16:02:16"),
-            np.datetime64("2005-01-04T03:14:12"),
-            np.datetime64("2041-12-03T14:05:03"),
-        ]
-    )
+DATETIME_INPUT = [
+    np.datetime64("1923-04-14T12:43:12"),
+    np.datetime64("1994-06-21T14:43:15"),
+    np.datetime64("2001-10-15T04:10:32"),
+    np.datetime64("NaT"),
+    np.datetime64("1995-11-25T16:02:16"),
+    np.datetime64("2005-01-04T03:14:12"),
+    np.datetime64("2041-12-03T14:05:03"),
+]
+
+
+TIMEDELTA_INPUT = [
+    np.timedelta64(12358, "s"),
+    np.timedelta64(23, "s"),
+    np.timedelta64(74, "s"),
+    np.timedelta64("NaT"),
+    np.timedelta64(23, "s"),
+    np.timedelta64(73, "s"),
+    np.timedelta64(7, "s"),
+]
+
+
+@pytest.mark.parametrize(
+    "input_data, input_dtype",
+    [
+        (DATETIME_INPUT, "M8[s]"),
+        (TIMEDELTA_INPUT, "m8[s]")
+    ]
+)
+def test_datetime_timedelta_cast(dtype, input_data, input_dtype):
+
+    a = np.array(input_data, dtype=input_dtype)
 
     has_na = hasattr(dtype, "na_object")
     is_str = isinstance(getattr(dtype, "na_object", None), str)
```

Comment:
I think it would be good to test NaT also with a default string of `NaT` (could be done in the next test too).

More generally, `NaT` should round-trip except when the default string is used and is not equal to "NaT"